LeaderboardBackend/Services/Impl/AccountRecoveryService.cs

```diff
@@ -70,4 +72,46 @@ private string GenerateAccountRecoveryEmailBody(User user, AccountRecovery recov
 
         return $@"Hi {user.Username},<br/><br/>Click <a href=""{builder.Uri}"">here</a> to reset your password.";
     }
+
+    public async Task<RecoverAccountResult> TestRecovery(Guid id)
+    {
+        AccountRecovery? recovery = await _applicationContext.AccountRecoveries.Include(ar => ar.User).SingleOrDefaultAsync(ar => ar.Id == id);
+
+        if (recovery is null)
+        {
+            return new NotFound();
+        }
+
+        if (recovery.User.Role is not UserRole.Registered)
+        {
+            return new BadRole();
```

Comment:
According to the spec, only the `Banned` role should prevent account recovery. But you probably would have noticed this if the tests were actually running :)